cmd/basic/main.go

```diff
@@ -78,152 +82,204 @@ func main() {
 		}
 		pprof.StartCPUProfile(f)
 		defer pprof.StopCPUProfile()
-		// TODO: Use a real timer with channel.
-		now := time.Now()
-		bl.stopAfter = now.Add(time.Second * 10)
-		fmt.Printf("time: %v\n", bl.stopAfter)
 	}
 
-	bl.Run()
+	testConfig.Run()
 }
 
-// Basic persistent load test, using text protocol:
-// - list of servers to connect to, pct of each.
-// - zipf or uniform random
-// - requests per connect (-1 for unlim)
-// - gets per etc
-// - multiget or not
-// - set or add to replace
-// - delete frequency
-// - set size range
-// - variances: peak/antipeak load
-// - variances: how often to change item sizes
-type BasicLoader struct {
-	servers               []string
-	stopAfter             time.Time
-	desiredConnCount      int
-	requestsPerSleep      int
-	requestBundlesPerConn int
-	sleepPerBundle        time.Duration
-	setValueSizes         []int
-	deletePercent         int
-	keyLength             int
-	keyPrefix             string
-	keySpace              int
-	keyTTL                uint
-	useZipf               bool
-	zipfS                 float64 // (> 1, generally 1.01-2) pulls the power curve toward 0)
-	zipfV                 float64 // v (< keySpace) puts the main part of the curve before this number
-	valueSize             uint
-	clientFlags           uint
+type Config struct {
+	ClientFlags    uint
+	ConnCount      int
+	DelRatio       int
+	Duration       time.Duration
+	GetRatio       int
+	KeyLength      int
+	KeyPrefix      string
+	KeySpace       int
+	KeyTTL         uint
+	Pipelines      uint
+	RngSeed        int64
+	RPS            int
+	Servers        []string
+	SetRatio       int
+	Socket         string
+	StripKeyPrefix bool
+	UseZipf        bool
+	ValueSize      uint
+	WarmPercent    int
+	ZipfS          float64 // (> 1, generally 1.01-2) pulls the power curve toward 0)
+	ZipfV          float64 // v (< keySpace) puts the main part of the curve before this number
+	tachymeter     *tachymeter.Tachymeter
```

Comment:
Removed the comment as it no longer reflected the code accurately and the features are documented in the flags. 

Renamed `BasicLoader` to `Config`.

Fields are sorted and exported so that they appear in the JSON report.